### ISSUE TITLE
Custom [Gmail]/Sent Items Label becomes inconsistent with System Sent Label, Flashes Messages in Sent Label

### ISSUE THREAD

**Issue by pdaniell** [Foundry376/Mailspring#1244]
##### Are there any related issues?

Yes. Issue 335; this issue says that Mailspring spawns an extra gmail "custom label" --- this bug concerns the fact that the Custom version of sent grows out of sync with the system label. 

...

##### What operating system are you using?

Ubuntu 16.04

...

##### What version of Mailspring are you using?

Version 1.5.3

...

--

**Bug?**

##### Is this an email sync or connection issue?

No
...

##### Is the issue related to a specific email provider (Gmail, Exchange, etc.)?

Yes Gmail. 
...

##### Is the issue reproducible with a particular attachment, message, signature, etc?

Yes, it's reproducible.
...

--

**Feature Request?**

##### Does this feature exist in another mail client or tool you use?

The Gmail Android, iOS and web interfaces can access the system labels. 

Description of the bug: 
It appears that "[Gmail]/Sent Mail" and "[Gmail]/Drafts" are custom labels created by Mailspring, but actually there are Gmail "system" labels called Sent and Drafts which appear to be canonical. I think the custom sent mail label is misapplied and hence the custom label gets out of sync with the system Sent label. Un my experience it the items in the custom "Sent Items" label are a subset of the ones in the Sent label. As in some comments in issue 335, it appears I cannot see the System "Sent" and System "Drafts" labels at all via Mailspring even though are marked as "show in IMAP.." I would be nice to see the canonical version of Sent emails using the system label.

...

**Comment by bengotow** [comment]
Hey! Thanks for reporting this. Mailspring should definitely be using the standard Gmail labels for Sent and Drafts. We've had a few other reports of this, and so far it seems that Gmail itself is creating the labels when you connect to your account via a third party IMAP client. Mailspring doesn't actually send IMAP commands to create the folders, it just tries to move mail into the existing ones and the duplicate folders are created.

So far I haven't had much luck narrowing this down. Do you use the Gmail web interface in another language by any chance? Maybe the drafts folder is not called "Drafts" when you look at Gmail webmail? Do you know if you're a particularly old Gmail user or a Google Apps for Work user? (Just trying to think of things that might narrow it down a bit.)

Could you also check that if you go to Gmail's preferences and look at the "IMAP" section you see that the Sent and Draft folders are "Enabled for IMAP"?

Thanks! Hopefully we can get this fixed soon.

**Comment by pdaniell** [comment]
(1) I don't use Gmail in any other language than English via the web. 
(2) I attach a picture of my label settings -- as you can see most of them are accessible via IMAP.. I actually have two different accounts --- one is a regular Gmail account and another is a Google Apps account --- but the label settings are indistinguishable except that my Apps account has a label called "[Gmail]/Drafts" where as the normal gmail does not. 
(3) I do think I was a quiet early user of gmail; I don't think my Google Apps account was particularly early though.

I did use Nylas before I used Mailspring and I believe Nylas created some labels -- at least I recall that it created a "Snoozed" label --- like Mailspring does. I am going to try disabling the custom labels in IMAP and see if that fixes anything.



![labels](https://user-images.githubusercontent.com/5911713/50209029-aa4b7780-036a-11e9-9c45-1af72e925be7.png)

**Comment by pdaniell** [comment]
@bengotow I actually now think that it's not that the messages  in Sent are merely inconsistent but that messages flash in and out of existence. The issue #1315 seems to report this behavior as well. I looked at the sync logs while this was happening and this is what I saw (see bottom). At the moment, this is only occurring in one of my two gmail accounts, though their "label settings" are identical (See screenshot in comment above). I'm not sure I understand the logs completely, but it looks like there are bunch of statements like

XGMLabels (["\\Sent"] to [])

in a block. (Does that mean "remove sent label", perhaps? Then there are of other statements like, 

 XGMLabels ([] to ["\\Sent"])

which seems to suggest that something is being put back in the Sent label. If a bunch of messages are being labeled and then unlabeled, that is perhaps why there is flashing?


```

26052 [2019-01-12 22:33:10.310] [background] [error] Could not create Mailspring container folder: Mailspring. ErrorCreate
26052 [2019-01-12 22:33:10.426] [background] [error] Could not create required Mailspring folder: Mailspring/Snoozed. ErrorCreate
26052 [2019-01-12 22:33:10.550] [background] [info] syncFolderUIDRange for [Gmail]/All Mail, UIDs: 24636 - 25312, Heavy: false
26052 [2019-01-12 22:33:11.081] [background] [info] - remote=432, local=400
26052 [2019-01-12 22:33:11.082] [background] [info] - Fetching full headers for 32 (of 32 needed)
26052 [2019-01-12 22:33:11.251] [background] [info] - Updating message Hryn9sNdtTzxawUYDj5KkRU8hqF4RUpQKgVEcW63C
26052 [2019-01-12 22:33:11.251] [background] [info] -- UID (6470 to 25308)
26052 [2019-01-12 22:33:11.251] [background] [info] -- FolderID (jAkk9NR5AKbmZU7G6QeqdTSTHnmrBaz1thhQptjRN to apFW7hDmkvXHj86CafsEjt3EbJQUjJ4Pu4ftTY7zb)
26052 [2019-01-12 22:33:11.251] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.255] [background] [info] - Updating message q6UVc5dSKD985mED6xhy3Xgp3YxZmbi2CjonRvzN8
26052 [2019-01-12 22:33:11.255] [background] [info] -- UID (6466 to 25269)
26052 [2019-01-12 22:33:11.255] [background] [info] -- FolderID (jAkk9NR5AKbmZU7G6QeqdTSTHnmrBaz1thhQptjRN to apFW7hDmkvXHj86CafsEjt3EbJQUjJ4Pu4ftTY7zb)
26052 [2019-01-12 22:33:11.255] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.258] [background] [info] - Updating message g6aosyNCWaXCenCn9g7PZqDvBNnn2tPwnxwtsPQP3
26052 [2019-01-12 22:33:11.258] [background] [info] -- UID (6465 to 25267)
26052 [2019-01-12 22:33:11.258] [background] [info] -- FolderID (jAkk9NR5AKbmZU7G6QeqdTSTHnmrBaz1thhQptjRN to apFW7hDmkvXHj86CafsEjt3EbJQUjJ4Pu4ftTY7zb)
26052 [2019-01-12 22:33:11.258] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.261] [background] [info] - Updating message QXNiDB3JHQaxutxSN1bqvong7LVdEdSzTgbMVRTfD
26052 [2019-01-12 22:33:11.261] [background] [info] -- UID (6452 to 25239)
26052 [2019-01-12 22:33:11.261] [background] [info] -- FolderID (jAkk9NR5AKbmZU7G6QeqdTSTHnmrBaz1thhQptjRN to apFW7hDmkvXHj86CafsEjt3EbJQUjJ4Pu4ftTY7zb)
26052 [2019-01-12 22:33:11.261] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.265] [background] [info] - Updating message jri7qV8HmP9UPhkhEsSDWyaMfMqTBuHGPaHH216RT
26052 [2019-01-12 22:33:11.265] [background] [info] -- UID (6445 to 25216)
26052 [2019-01-12 22:33:11.265] [background] [info] -- FolderID (jAkk9NR5AKbmZU7G6QeqdTSTHnmrBaz1thhQptjRN to apFW7hDmkvXHj86CafsEjt3EbJQUjJ4Pu4ftTY7zb)
26052 [2019-01-12 22:33:11.265] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.268] [background] [info] - Updating message qHw7ZEEASTRxUXZ9bpfVCCxyQTLukbFw6jmmTVJNQ
26052 [2019-01-12 22:33:11.268] [background] [info] -- UID (6444 to 25209)
26052 [2019-01-12 22:33:11.268] [background] [info] -- FolderID (jAkk9NR5AKbmZU7G6QeqdTSTHnmrBaz1thhQptjRN to apFW7hDmkvXHj86CafsEjt3EbJQUjJ4Pu4ftTY7zb)
26052 [2019-01-12 22:33:11.268] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.271] [background] [info] - Updating message MnMAXTQv3v1rfqfahKp4gMf6xwyV6Bjpsvv1uaotE
26052 [2019-01-12 22:33:11.271] [background] [info] -- UID (6432 to 25169)
26052 [2019-01-12 22:33:11.271] [background] [info] -- FolderID (jAkk9NR5AKbmZU7G6QeqdTSTHnmrBaz1thhQptjRN to apFW7hDmkvXHj86CafsEjt3EbJQUjJ4Pu4ftTY7zb)
26052 [2019-01-12 22:33:11.271] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.275] [background] [info] - Updating message 5mkbJsRQBK5N4CnvnLmdPqotPhFqzejxXzCoZoUAr
26052 [2019-01-12 22:33:11.275] [background] [info] -- UID (24866 to 24922)
26052 [2019-01-12 22:33:11.275] [background] [info] -- XGMLabels (["\\Sent"] to ["\\Inbox"])
26052 [2019-01-12 22:33:11.278] [background] [info] - Updating message b5kWTiqwq6oMWJSFGD4G3jbeYvdbEz7r8cEZNHqqr
26052 [2019-01-12 22:33:11.278] [background] [info] -- UID (24736 to 24737)
26052 [2019-01-12 22:33:11.278] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.281] [background] [info] - Updating message EcnuyK6GetT5u2C4sVf8o5PyDjB4AQ6cEJ3hXKaAQ
26052 [2019-01-12 22:33:11.281] [background] [info] -- UID (24734 to 24735)
26052 [2019-01-12 22:33:11.281] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.284] [background] [info] - Updating message R1NGhheEiy6LBcx3cQFUriFvQmF6de4ETDRJG6L3N
26052 [2019-01-12 22:33:11.284] [background] [info] -- UID (24732 to 24733)
26052 [2019-01-12 22:33:11.284] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.287] [background] [info] - Updating message GeA8keJFCDNCr5pySbBA6xfckc2FpaUER8YUQB8Q
26052 [2019-01-12 22:33:11.287] [background] [info] -- UID (24730 to 24731)
26052 [2019-01-12 22:33:11.287] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.291] [background] [info] - Updating message nqgukanKiGdhNNDtdqxQ9ACLE4X4naHRBRPi1WiDr
26052 [2019-01-12 22:33:11.291] [background] [info] -- UID (24728 to 24729)
26052 [2019-01-12 22:33:11.291] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.294] [background] [info] - Updating message 9sTxKtGKh23gepFffEZx94r5JVXkTZYApuPWCWDwn
26052 [2019-01-12 22:33:11.294] [background] [info] -- UID (24726 to 24727)
26052 [2019-01-12 22:33:11.294] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.297] [background] [info] - Updating message CiHgK9PU1p1AL7uuaioC2bjWDWcrzd74y9vBiG4Xh
26052 [2019-01-12 22:33:11.297] [background] [info] -- UID (24724 to 24725)
26052 [2019-01-12 22:33:11.297] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.300] [background] [info] - Updating message EwQqcqDq4PVPpKUPqopZxgj4yLePKq1dcvcGk9Lr7
26052 [2019-01-12 22:33:11.300] [background] [info] -- UID (24722 to 24723)
26052 [2019-01-12 22:33:11.300] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.303] [background] [info] - Updating message 5BvXNa4Pc7uwGQLJ7ud8sapdDJyKzCg234pWEGHnN
26052 [2019-01-12 22:33:11.303] [background] [info] -- UID (24720 to 24721)
26052 [2019-01-12 22:33:11.304] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.307] [background] [info] - Updating message 5tWc9BU1SH8FZKss1nybwCaz3RBvXfLwB91ohGrvo
26052 [2019-01-12 22:33:11.307] [background] [info] -- UID (24718 to 24719)
26052 [2019-01-12 22:33:11.307] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.310] [background] [info] - Updating message iKSxMxV4opiLpnWR6QwPiS8cqLu6JA7ZYy2EEvacv
26052 [2019-01-12 22:33:11.310] [background] [info] -- UID (24716 to 24717)
26052 [2019-01-12 22:33:11.310] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.333] [background] [info] - Updating message sp7nVePZfVRt1gNBzMC6HwmUCZmTAYLUnWNcXYsVm
26052 [2019-01-12 22:33:11.333] [background] [info] -- UID (24714 to 24715)
26052 [2019-01-12 22:33:11.333] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.342] [background] [info] - Updating message eHoUDeUSXvxmwaX4Lph5SsYo6iFMrmXU3aFH76wSA
26052 [2019-01-12 22:33:11.342] [background] [info] -- UID (24712 to 24713)
26052 [2019-01-12 22:33:11.342] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.347] [background] [info] - Updating message Kn9Rq8MgD2rUMRgSuKfyvjaaa3BZ7nbby4aXDqh5C
26052 [2019-01-12 22:33:11.347] [background] [info] -- UID (24710 to 24711)
26052 [2019-01-12 22:33:11.347] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.350] [background] [info] - Updating message QGEswgT2ssrBCx6vGr7bauszpKReeQ2o1PR7s1pSu
26052 [2019-01-12 22:33:11.350] [background] [info] -- UID (24708 to 24709)
26052 [2019-01-12 22:33:11.350] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.354] [background] [info] - Updating message 85wdaEDZVzZgiPbT2hy4KqLRYHFzSWyR82hpNQvTd
26052 [2019-01-12 22:33:11.354] [background] [info] -- UID (24706 to 24707)
26052 [2019-01-12 22:33:11.354] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.356] [background] [info] - Updating message 8igDJHQy1w3bXM6oDstDGy4VZNHoGLpxm9iaGPK4X
26052 [2019-01-12 22:33:11.357] [background] [info] -- UID (24704 to 24705)
26052 [2019-01-12 22:33:11.357] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.359] [background] [info] - Updating message NJmhMg3BeYJ9Hzmx4ZcJruvUiFakKEiNiiXcyGacv
26052 [2019-01-12 22:33:11.359] [background] [info] -- UID (24702 to 24703)
26052 [2019-01-12 22:33:11.360] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.362] [background] [info] - Updating message NuPGxeDHgqzTzK8bREKot3BP3rgw7XEKStUHMV3p7
26052 [2019-01-12 22:33:11.363] [background] [info] -- UID (24700 to 24701)
26052 [2019-01-12 22:33:11.363] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.366] [background] [info] - Updating message ZovVQVrwfmq6fvy69KcCZX2avfAxRyvfdpe1oQL3e
26052 [2019-01-12 22:33:11.366] [background] [info] -- UID (24698 to 24699)
26052 [2019-01-12 22:33:11.366] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.369] [background] [info] - Updating message bRPqe7CzJyZ21BRLWKcn1HicbMrnSMKKcAL1RogPe
26052 [2019-01-12 22:33:11.369] [background] [info] -- UID (24696 to 24697)
26052 [2019-01-12 22:33:11.369] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.372] [background] [info] - Updating message fPoj2m4fGyTSY14564vj199WiuyxEcAQ9J4wgaXdk
26052 [2019-01-12 22:33:11.372] [background] [info] -- UID (24694 to 24695)
26052 [2019-01-12 22:33:11.372] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.375] [background] [info] - Updating message nyfxaQTfyRbkWVUYw11tMsRccNGW8hMTDHUAGFYFL
26052 [2019-01-12 22:33:11.375] [background] [info] -- UID (24692 to 24693)
26052 [2019-01-12 22:33:11.375] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.378] [background] [info] - Updating message AXpdwXnNFu94rdUKNU8GVfon32NmTBmmWWsdSxQwx
26052 [2019-01-12 22:33:11.378] [background] [info] -- UID (24690 to 24691)
26052 [2019-01-12 22:33:11.378] [background] [info] -- XGMLabels ([] to ["\\Sent"])
26052 [2019-01-12 22:33:11.542] [background] [info] syncFolderUIDRange for [Gmail]/Trash, UIDs: 1 - 6471, Heavy: false
26052 [2019-01-12 22:33:11.799] [background] [info] - remote=7, local=0
26052 [2019-01-12 22:33:11.799] [background] [info] - Fetching full headers for 7 (of 7 needed)
26052 [2019-01-12 22:33:11.945] [background] [info] - Updating message Hryn9sNdtTzxawUYDj5KkRU8hqF4RUpQKgVEcW63C
26052 [2019-01-12 22:33:11.945] [background] [info] -- UID (25308 to 6470)
26052 [2019-01-12 22:33:11.945] [background] [info] -- FolderID (apFW7hDmkvXHj86CafsEjt3EbJQUjJ4Pu4ftTY7zb to jAkk9NR5AKbmZU7G6QeqdTSTHnmrBaz1thhQptjRN)
26052 [2019-01-12 22:33:11.945] [background] [info] -- XGMLabels (["\\Sent"] to [])
26052 [2019-01-12 22:33:11.951] [background] [info] - Updating message q6UVc5dSKD985mED6xhy3Xgp3YxZmbi2CjonRvzN8
26052 [2019-01-12 22:33:11.951] [background] [info] -- UID (25269 to 6466)
26052 [2019-01-12 22:33:11.951] [background] [info] -- FolderID (apFW7hDmkvXHj86CafsEjt3EbJQUjJ4Pu4ftTY7zb to jAkk9NR5AKbmZU7G6QeqdTSTHnmrBaz1thhQptjRN)
26052 [2019-01-12 22:33:11.951] [background] [info] -- XGMLabels (["\\Sent"] to [])
26052 [2019-01-12 22:33:11.956] [background] [info] - Updating message g6aosyNCWaXCenCn9g7PZqDvBNnn2tPwnxwtsPQP3
26052 [2019-01-12 22:33:11.956] [background] [info] -- UID (25267 to 6465)
26052 [2019-01-12 22:33:11.956] [background] [info] -- FolderID (apFW7hDmkvXHj86CafsEjt3EbJQUjJ4Pu4ftTY7zb to jAkk9NR5AKbmZU7G6QeqdTSTHnmrBaz1thhQptjRN)
26052 [2019-01-12 22:33:11.956] [background] [info] -- XGMLabels (["\\Sent"] to [])
26052 [2019-01-12 22:33:11.959] [background] [info] - Updating message QXNiDB3JHQaxutxSN1bqvong7LVdEdSzTgbMVRTfD
26052 [2019-01-12 22:33:11.959] [background] [info] -- UID (25239 to 6452)
26052 [2019-01-12 22:33:11.959] [background] [info] -- FolderID (apFW7hDmkvXHj86CafsEjt3EbJQUjJ4Pu4ftTY7zb to jAkk9NR5AKbmZU7G6QeqdTSTHnmrBaz1thhQptjRN)
26052 [2019-01-12 22:33:11.959] [background] [info] -- XGMLabels (["\\Sent"] to [])
26052 [2019-01-12 22:33:11.962] [background] [info] - Updating message jri7qV8HmP9UPhkhEsSDWyaMfMqTBuHGPaHH216RT
26052 [2019-01-12 22:33:11.962] [background] [info] -- UID (25216 to 6445)
26052 [2019-01-12 22:33:11.962] [background] [info] -- FolderID (apFW7hDmkvXHj86CafsEjt3EbJQUjJ4Pu4ftTY7zb to jAkk9NR5AKbmZU7G6QeqdTSTHnmrBaz1thhQptjRN)
26052 [2019-01-12 22:33:11.962] [background] [info] -- XGMLabels (["\\Sent"] to [])
26052 [2019-01-12 22:33:11.965] [background] [info] - Updating message qHw7ZEEASTRxUXZ9bpfVCCxyQTLukbFw6jmmTVJNQ
26052 [2019-01-12 22:33:11.965] [background] [info] -- UID (25209 to 6444)
26052 [2019-01-12 22:33:11.965] [background] [info] -- FolderID (apFW7hDmkvXHj86CafsEjt3EbJQUjJ4Pu4ftTY7zb to jAkk9NR5AKbmZU7G6QeqdTSTHnmrBaz1thhQptjRN)
26052 [2019-01-12 22:33:11.965] [background] [info] -- XGMLabels (["\\Sent"] to [])
26052 [2019-01-12 22:33:11.968] [background] [info] - Updating message MnMAXTQv3v1rfqfahKp4gMf6xwyV6Bjpsvv1uaotE
26052 [2019-01-12 22:33:11.968] [background] [info] -- UID (25169 to 6432)
26052 [2019-01-12 22:33:11.968] [background] [info] -- FolderID (apFW7hDmkvXHj86CafsEjt3EbJQUjJ4Pu4ftTY7zb to jAkk9NR5AKbmZU7G6QeqdTSTHnmrBaz1thhQptjRN)
26052 [2019-01-12 22:33:11.968] [background] [info] -- XGMLabels (["\\Sent"] to [])
26052 [2019-01-12 22:33:11.970] [background] [info] Sync loop deleting unlinked messages with phase 1.
26052 [2019-01-12 22:33:11.975] [background] [info] Sync loop complete.
``

**Comment by bengotow** [comment]
Hey folks—this should be fixed in the next release by a4676d730d01ac132accf2463e94fdb78eebe10c. Stay tuned!

**Comment by bengotow** [comment]
Hey folks! This should be addressed in the latest release (1.6.3) that shipped earlier this week. The app should auto-update on it's own, or you can grab the new version from the releases page: https://github.com/Foundry376/Mailspring/releases/. Thanks for filing this - enjoy the update!
